File: dsa/li01_array/L1752_CheckifArrayIsSortedandRotated.java
package li01_array;

public class L1752_CheckifArrayIsSortedandRotated {

    public static void main(String[] args) {
        L1752_CheckifArrayIsSortedandRotated obj = new L1752_CheckifArrayIsSortedandRotated();
        int[] arr = { 3, 4, 5, 1, 2 };
        System.out.println(obj.check(arr));
    }

    // Problem: https://leetcode.com/problems/check-if-array-is-sorted-and-rotated/
    public boolean check(int[] nums){
        int n = nums.length;
        int count = 0;
        if(nums[0] < nums[n-1])
            count++;
        for(int i=1; i<n; i++){
            if(nums[i] < nums[i-1]){
                count++;
            }
        }

        return count <= 1;
    }

}

File: dsa/patterns/01_arrays/python/l1752_check_if_array_is_sorted_and_rotated.py
"""
L1752 - Check if Array Is Sorted and Rotated

Problem: https://leetcode.com/problems/check-if-array-is-sorted-and-rotated/
"""

def check(nums):
    n = len(nums)
    count = 0
    
    if nums[0] < nums[n - 1]:
        count += 1
    
    for i in range(1, n):
        if nums[i] < nums[i - 1]:
            count += 1
    
    return count <= 1


if __name__ == "__main__":
    arr = [3, 4, 5, 1, 2]
    print(check(arr))
